build.rs

```diff
@@ -18,21 +22,77 @@ fn main() {
         let lapacke = feature!("LAPACKE");
         let source = PathBuf::from("source");
         let output = PathBuf::from(variable!("OUT_DIR").replace(r"\", "/"));
+        // The target triple provided by cargo --target is not compatible with
+        // OpenBLAS targets.
         env::remove_var("TARGET");
-        run(
-            Command::new("make")
-                .args(&["libs", "netlib", "shared"])
-                .arg(format!("BINARY={}", binary!()))
-                .arg(format!("{}_CBLAS=1", switch!(cblas)))
-                .arg(format!("{}_LAPACKE=1", switch!(lapacke)))
-                .arg(format!("-j{}", variable!("NUM_JOBS")))
-                .current_dir(&source),
-        );
+        // The working directory for both 'make' and 'make install'
+        let make_working_dir: PathBuf;
```

Comment:
Am I right that you always have to pass `--target` to Cargo along with those three environment variables? If so, what do you say if we always compile in a directory called `TARGET` (the environment variable) so as to unify the treatment of cross and non-cross compilation?

Reply:
Let’s rename it to `target`.

Reply:
I switched to the unified approach. I didn't originally because it's just a tad inconsistent in that the target folder names are now a mix of the cargo target (one space of options) and openblas targets (another space of options).

Reply:
Just for my understanding: what is `TARGET` equal to if one does cross compilation? If it’s the actual target platform but in a format (the usual triple) that is not suitable for OpenBLAS, why wouldn’t it be appropriate to just use `TARGET` for naming the working directory instead of alternating between `TARGET` and `OPENBLAS_TARGET`?

Reply:
I can't say for certain there there would be a problem with what you're suggesting. But, there's one slight concern. In your understanding, there's a direct one-to-one mapping between cargo targets and openblas targets, even if their names differ. For example: armv7-unknown-linux-gnueabihf -> armv7.

However, for the cargo triple x86_64-unknown-linux-gnu, there are lots of openblas targets that are possible. You can see the list [here](https://github.com/xianyi/OpenBLAS/blob/develop/TargetList.txt).

`rustc` has this information too (`rustc --print target-cpus`) but AFAIK does not pass this information to openblas; I looked through the full list of environment variables being passed. Thus, it's possible for the same cargo target to be used for multiple openblas targets.

Will users realistically do this? Maybe if they're testing performance across slightly different instruction sets. The user will also need to be savvy enough to run `cargo clean` since changing `OPENBLAS_TARGET` won't result in a recompilation.

Anyway, those are just my thoughts. Happy to do it either way.